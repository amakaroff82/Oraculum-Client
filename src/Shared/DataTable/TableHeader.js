import React from 'react';
import { Grid, Toolbar } from 'material-ui';
import Typography from 'material-ui/Typography';
import { withStyles } from 'material-ui/styles';

const styles = theme => ({
  root: {
    height: theme.spacing.unit * 12,
  },
});

export const TableHeader = ({
  classes,
  headerLabel,
  leftButtons,
  rightButtons,
}) => {
  return (
    <Grid container alignItems="center" wrap="nowrap" className={classes.root}>
      <Grid item xs={6}>
        <Toolbar>
          <Typography type="title">{headerLabel}</Typography>
          <Toolbar>{leftButtons}</Toolbar>
        </Toolbar>
      </Grid>
      <Grid item xs={6}>
        <Grid container justify="flex-end">
          <Grid item>
            <Toolbar>{rightButtons}</Toolbar>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
};

export default withStyles(styles, { name: 'TableHeader' })(TableHeader);
