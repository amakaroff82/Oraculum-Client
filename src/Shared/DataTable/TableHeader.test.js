import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { TableHeader } from './TableHeader';

const defaultProps = {
  classes: {},
  headerLabel: 'Test',
  leftButtons: [<div>{'TestLButton'}</div>],
  rightButtons: [<div>{'TestRButton1'}</div>, <div>{'TestRButton2'}</div>],
};

const setup = props => {
  props = { ...defaultProps, ...props };

  return shallow(<TableHeader {...props} />);
};

describe('<TableHeader />', () => {
  it('renders snapshot correctly with required props', () => {
    const tree = setup();
    expect(toJson(tree)).toMatchSnapshot();
  });
});
