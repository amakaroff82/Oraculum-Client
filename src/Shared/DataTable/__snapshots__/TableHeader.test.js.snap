// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableHeader /> renders snapshot correctly with required props 1`] = `
<withStyles(GridWrapper)
  alignItems="center"
  container={true}
  wrap="nowrap"
>
  <withStyles(GridWrapper)
    item={true}
    xs={6}
  >
    <withStyles(Toolbar)>
      <withStyles(Typography)
        type="title"
      >
        Test
      </withStyles(Typography)>
      <pure(ChevronRight)
        viewBox="0 0 120 120"
      />
      <withStyles(Toolbar)>
        <div>
          TestLButton
        </div>
      </withStyles(Toolbar)>
    </withStyles(Toolbar)>
  </withStyles(GridWrapper)>
  <withStyles(GridWrapper)
    item={true}
    xs={6}
  >
    <withStyles(GridWrapper)
      container={true}
      justify="flex-end"
    >
      <withStyles(GridWrapper)
        item={true}
      >
        <withStyles(Toolbar)>
          <div>
            TestRButton1
          </div>
          <div>
            TestRButton2
          </div>
        </withStyles(Toolbar)>
      </withStyles(GridWrapper)>
    </withStyles(GridWrapper)>
  </withStyles(GridWrapper)>
</withStyles(GridWrapper)>
`;
